knock

```diff
@@ -30,22 +30,33 @@ class Knocker(object):
         self.timeout = args.timeout / 1000
         self.delay = args.delay / 1000
         self.use_udp = args.udp
-        self.host = args.host
-        self.is_ipv6 = args.host.find(':') != -1
         self.ports = args.ports
 
+        self.address_family, self.socket_type, _, _, socket_address = socket.getaddrinfo(
+                host=args.host,
+                port=None,
+                type=0,
+                proto=socket.IPPROTO_UDP if self.use_udp else socket.IPPROTO_TCP,
```

Comment:
Strange ... alright, I changed it to filter by `type` instead - I'm happy that it works for you. Thank you very much for the thorough testing :)